cpu/stm32/periph/flashpage.c

```diff
@@ -212,17 +249,35 @@ void flashpage_write(void *target_addr, const void *data, size_t len)
     stmclk_enable_hsi();
 #endif
 
+#ifdef FLASH_ACR_DCEN
+    if (FLASH->ACR & FLASH_ACR_DCEN) {
+        FLASH->ACR &= ~FLASH_ACR_DCEN;
+    }
+#endif
+#ifdef FLASH_ACR_ICEN
+    if (FLASH->ACR & FLASH_ACR_ICEN) {
+        FLASH->ACR &= ~FLASH_ACR_ICEN;
+    }
+#endif
+
     /* unlock the flash module */
     _unlock_flash();
 
+#if defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4) || \
+    defined(CPU_FAM_STM32F7)
+    CNTRL_REG |= FLASH_CR_PSIZE_1; /* Word size parallelism */
```

Comment:
According to the manual: 
```
The Parallelism size is configured through the PSIZE field in the FLASH_CR register. It
represents the number of bytes to be programmed each time a write operation occurs to the
Flash memory
```
But this should be made configurable IMO because this setting depends on the supply voltage:
```
PSIZE is limited by the supply voltage and by whether the external VPP
supply is used or not. It must therefore be correctly configured in the FLASH_CR register
before any programming/erasing operation. 
```

Reply:
I agree, but I would really like to leave this for a follow up. It makes the current significantly more complex because the correct data size for copying to the flash has to be selected at compile time.

I can document that this currently only works from 2.7V - 3.3V

Reply:
Should be simpler to add with #15467 merged